### PR TITLE
Expect.expectCollection, new ExpectMap and test cleanups

### PR DESCRIPTION
**Pull request by h2000**

### CODE REVIEW

File: src/main/java/de/tobiasroeser/lambdatest/Expect.java [hunk 1]
```diff
@@ -223,6 +224,30 @@ public static ExpectString expectString(final String actual) {
...
+	 * @param actual
+	 *            The Collection<T> to check.
+	 * @return A {@link ExpectCollection} to express further expectations on the
+	 *         actual collection.
+	 */
+	public static <T> ExpectCollection<T> expectCollection(final Collection<T> actual) {
+		return new ExpectCollection<>(actual);
+	}
+
+	public static ExpectString expect(final String actual) {
```
[the remark this answers is not in the document]
Comment (h2000) [reply src/main/java/de/tobiasroeser/lambdatest/Expect.java]: The motivation comes from assertjs assertThat(..) method. 

```java
// entry point for all assertThat methods and utility methods (e.g. entry)
import static org.assertj.core.api.Assertions.*;

// basic assertions
assertThat(frodo.getName()).isEqualTo("Frodo");
assertThat(frodo).isNotEqualTo(sauron);

// chaining string specific assertions
assertThat(frodo.getName()).startsWith("Fro")
                           .endsWith("do")
                           .isEqualToIgnoringCase("frodo");

// collection specific assertions (there are plenty more)
// in the examples below fellowshipOfTheRing is a List<TolkienCharacter>
assertThat(fellowshipOfTheRing).hasSize(9)
                               .contains(frodo, sam)
                               .doesNotContain(sauron);
```

[the remark this answers is not in the document]
Comment (h2000) [reply src/main/java/de/tobiasroeser/lambdatest/Expect.java]: Overloads of expect(Collection<T>) and expect(Map<K,V>) are no problem, see current commit in h2000:master

[the remark this answers is not in the document]
Comment (h2000) [reply src/main/java/de/tobiasroeser/lambdatest/Expect.java]: Intellij says Arrays.asList is  not explicit enough for places of empty collection or singled lists.

**Comment by lefou** [comment]
Thank you!

**Pull request merged**